src/Tools/dotnet-monitor/CollectionRules/Actions/ICollectionRuleAction.cs

```diff
@@ -2,19 +2,17 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using Microsoft.Diagnostics.Monitoring.WebApi;
-using Microsoft.Diagnostics.NETCore.Client;
-using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions
 {
-    internal interface ICollectionRuleAction<TOptions>
+    internal interface ICollectionRuleAction
```

Comment:
Even if this method did exist, it would likely be used as:

```cs
using IDisposable _ = token.Register(() => action.Cancel(token));
```

and the code would likely still be conditionally disposing the action.

Reply:
DisposeAsync essentially provides this functionality right?

Reply:
Correct.